src/nifgen/system_tests/test_system_nifgen.py

```diff
@@ -26,538 +29,576 @@
+            assert simulated_session.channel_count == 2
+        with nifgen.Session('', '0,1', False, 'Simulate=1, DriverSetup=Model:5433 (2CH);BoardType:PXIe') as simulated_session:
+            assert simulated_session.channel_count == 2
+        with nifgen.Session('', None, False, 'Simulate=1, DriverSetup=Model:5433 (2CH); BoardType:PXIe') as simulated_session:
+            assert simulated_session.channel_count == 2
+        with nifgen.Session(resource_name='', reset_device=False, options='Simulate=1, DriverSetup=Model:5433 (2CH); BoardType:PXIe') as simulated_session:
+            assert simulated_session.channel_count == 2
+
+    # Test doesn't run over gRPC because numpy isn't supported by gRPC.
+    def test_create_waveform_from_numpy_array_float64(self, session):
+        data = numpy.ndarray(10000, dtype=numpy.float64)
+        data.fill(0.5)
+        assert type(session.create_waveform(data)) is int
+
+    # Test doesn't run over gRPC because numpy isn't supported by gRPC.
+    def test_create_waveform_numpy_array_int16(self, session):
+        data = numpy.ndarray(10000, dtype=numpy.int16)
+        data.fill(256)
+        assert type(session.create_waveform(data)) is int
+
+    # Test doesn't run over gRPC because numpy isn't supported by gRPC.
+    def test_create_waveform_numpy_array_int16(self, session):
+        data = numpy.ndarray(10000, dtype=numpy.int16)
+        data.fill(256)
+        assert type(session.create_waveform(data)) is int
+
+    # Test doesn't run over gRPC because numpy isn't supported by gRPC.
+    def test_write_waveform_from_numpy_array_float64(self, session):
+        data = numpy.ndarray(10000, dtype=numpy.float64)
+        data.fill(0.5)
+        session.write_waveform(session.allocate_waveform(len(data)), data)
+
+    # Test doesn't run over gRPC because numpy isn't supported by gRPC.
+    def test_write_waveform_numpy_array_int16(self, session):
+        data = numpy.ndarray(10000, dtype=numpy.int16)
+        data.fill(256)
+        session.write_waveform(session.allocate_waveform(len(data)), data)
+
+    # Test doesn't run over gRPC because numpy isn't supported by gRPC.
+    def test_write_named_waveform_from_numpy_array_float64(self, session):
+        data = numpy.ndarray(10000, dtype=numpy.float64)
+        data.fill(0.5)
+        session.allocate_named_waveform('foo', len(data))
+        session.write_waveform('foo', data)
+
+    # Test doesn't run over gRPC because numpy isn't supported by gRPC.
+    def test_write_named_waveform_numpy_array_int16(self, session):
+        data = numpy.ndarray(10000, dtype=numpy.int16)
+        data.fill(256)
+        session.allocate_named_waveform('foo', len(data))
+        session.write_waveform('foo', data)
+
+    # Test doesn't run over gRPC because the message SetAttributeViBooleanRequest's attribute is called attribute_value rather than attribute_value_raw.
```

Comment:
This failure should be fixed. For this one, note that [my hapigen PR yesterday](https://ni.visualstudio.com/DevCentral/_git/ni-central/pullrequest/382146) should have the fix for it, so you should just need to grab the latest metadata.

Reply:
I know you have another PR up to fix this in hapigen following our further discussion on it, but I went ahead and manually fixed it in the `.proto` and `functions.py` for now.